contracts/mocks/ConvexYieldWrapperMock.sol

```diff
@@ -0,0 +1,351 @@
+
+    function withdrawAndUnwrap(uint256 amount, bool claim) external;
+
+    function earned(address account) external view returns (uint256);
+
+    function getReward() external;
+
+    function getReward(address _account, bool _claimExtras) external;
+
+    function extraRewardsLength() external view returns (uint256);
+
+    function extraRewards(uint256 _pid) external view returns (address);
+
+    function rewardToken() external view returns (address);
+
+    function balanceOf(address _account) external view returns (uint256);
+}
+
+contract ConvexYieldWrapperMock is ERC20, AccessControl {
+    using TransferHelper for IERC20;
+
+    struct RewardType {
+        address reward_token;
+        address reward_pool;
+        uint256 reward_integral;
+        uint256 reward_remaining;
+        mapping(address => uint256) reward_integral_for;
+        mapping(address => uint256) claimable_reward;
+    }
+
+    mapping(address => bytes12[]) public vaults;
+    ICauldron cauldron;
+
+    uint256 public cvx_reward_integral;
+    uint256 public cvx_reward_remaining;
+    mapping(address => uint256) public cvx_reward_integral_for;
+    mapping(address => uint256) public cvx_claimable_reward;
+
+    //constants/immutables
+    // address public constant convexBooster = address(0xF403C135812408BFbE8713b5A23a04b3D48AAE31);
+    address public crv;
+    address public cvx;
+    address public convexToken;
+    address public convexPool;
+    uint256 public convexPoolId;
+    address public collateralVault;
+
+    //rewards
+    RewardType[] public rewards;
+
+    event Deposited(address indexed _user, address indexed _account, uint256 _amount, bool _wrapped);
+    event Withdrawn(address indexed _user, uint256 _amount, bool _unwrapped);
+    event OwnershipTransferred(address indexed previousOwner, address indexed newOwner);
+
+    constructor(
+        address convexToken_,
+        address convexPool_,
+        uint256 poolId_,
+        address join_,
+        ICauldron cauldron_,
+        address crv_,
+        address cvx_
+    ) ERC20('StakedConvexToken', 'stkCvx', 18) {
+        convexToken = convexToken_;
+        convexPool = convexPool_;
+        convexPoolId = poolId_;
+        collateralVault = join_; //TODO: Add the join address
+        cauldron = cauldron_;
+        crv = crv_;
+        cvx = cvx_;
+        setApprovals();
+        addRewards();
+    }
+
+    function setCauldron(ICauldron _cauldron) external {
+        cauldron = _cauldron;
+    }
+
+    // Set the locations of vaults where the user's funds have been deposited & the accounting is kept
+    function addVault(bytes12 vault_) external {
+        address account = cauldron.vaults(vault_).owner;
+        require(account != address(0), 'No owner for the vault');
+        bytes12[] storage userVault = vaults[account];
+        for (uint256 i = 0; i < userVault.length; i++) {
+            require(userVault[i] != vault_, 'already added');
```

Comment:
Do we need to revert? Would be better to just return?

Both options are fine, but let's be consistent here and in `removeVault`, and consult with frontend engineering.